src/actions/userActions.ts

```diff
@@ -101,6 +103,22 @@ export const signOut = () => async (dispatch): Promise<void> => {
     dispatch(hideBackdrop())
 }
 
+export const signInAnonymously = () => async (dispatch) => {
+    dispatch(showBackdrop())
```

Comment:
I think it's handled in `AuthStateChangedListener`. That component listens for `auth` state. In case no user is passed to the callback, then we proceed to anonymous login. In case a user is passed (either anonymous or not), we go to `didSignIn` action in order to fetch the token

Reply:
I understand but it may be better to isolate the logic inside one action.